app/src/main/java/com/example/bigowlapp/utils/SupervisorSmsListener.java

```diff
@@ -0,0 +1,72 @@
+package com.example.bigowlapp.utils;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.telephony.SmsMessage;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.RequiresApi;
+
+import com.example.bigowlapp.model.Schedule;
+import com.google.firebase.Timestamp;
+
+import java.util.Calendar;
+
+@RequiresApi(api = Build.VERSION_CODES.O)
+public class SupervisorSmsListener extends BroadcastReceiver {
+
+    private static final String SMS_RECEIVED = "android.provider.Telephony.SMS_RECEIVED";
+    private String smsSender;
```

Comment:
this could be local variable, unless you need it for other purpose like testing

Reply:
Not at this point